internal/file/file.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package file
+
+import (
+	"time"
+
+	"github.com/uber/prototool/internal/x/settings"
+	"go.uber.org/zap"
+)
+
+// DefaultWalkTimeout is the default walk timeout.
+const DefaultWalkTimeout time.Duration = 3 * time.Second
+
+// ProtoSet represents a set of .proto files and an associated config.
+//
+// ProtoSets will be validated if returned from this package.
```

Comment:
Yea this is a little awkward, but I think `considered` is wrong - I'm saying that if you construct a ProtoSet from a function in this package, it will be pre-validated and you can assume they are valid, not considered to be validated...ugh maybe considered is the same thing, I don't know, thoughts?